app/utils/tests_import.py

```diff
@@ -450,38 +449,49 @@ def import_test_case(
     if isinstance(json_obj, types.DictionaryType):
         j_get = json_obj.get
         json_suite_id = j_get(models.TEST_SUITE_ID_KEY, None)
+        suite_oid = bson.objectid.ObjectId(suite_id)
 
         json_obj[models.TEST_SUITE_NAME_KEY] = suite_name
         if not json_suite_id:
             # Inject the suite_id value into the data structure.
             json_obj[models.TEST_SUITE_ID_KEY] = suite_id
         else:
             if json_suite_id == str(suite_id):
-                json_obj[models.TEST_SUITE_ID_KEY] = suite_id
+                json_obj[models.TEST_SUITE_ID_KEY] = suite_oid
             else:
                 utils.LOG.warning(
                     "Test suite ID does not match the provided one")
                 # XXX For now, force the suite_id value.
-                json_obj[models.TEST_SUITE_ID_KEY] = suite_id
+                json_obj[models.TEST_SUITE_ID_KEY] = suite_oid
 
         try:
             test_name = j_get(models.NAME_KEY, None)
+            set_id = j_get(models.TEST_SET_ID_KEY, None)
+            if set_id:
+                json_obj[models.TEST_SET_ID_KEY] = bson.objectid.ObjectId(set_id)
+            else:
+                json_obj[models.TEST_SET_ID_KEY] = None
             test_case = mtcase.TestCaseDocument.from_json(json_obj)
 
             if test_case:
-                k_get = kwargs.get
-
                 test_case.created_on = datetime.datetime.now(
                     tz=bson.tz_util.utc)
-                test_case.test_set_id = k_get(models.TEST_SET_ID_KEY, None)
-
                 ret_val, doc_id = utils.db.save(
                     database, test_case, manipulate=True)
 
                 if ret_val != 201:
                     err_msg = "Error saving test case '%s'" % test_name
                     utils.LOG.error(err_msg)
                     ADD_ERR(errors, 500, err_msg)
+                # If test case imported correctly reference it in the test suite
+                else:
+                    update_test_suite_add_test_case_id(
+                        doc_id, suite_oid, suite_name,
+                        db_options)
+                    #and in the test set if it exists
+                    if set_id:
+                        update_test_set_add_test_case_id(
+                            doc_id, bson.objectid.ObjectId(set_id), db_options)
```

Comment:
If you assign the `ObjectId` value to `set_id` before, you don't need to call `bson.objectid.ObjectId(set_id)` again here.